File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/db2/DomainType.kt
package dev.typr.foundationskt.docs.db2

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class DomainType {
    //start
    // Wrapper type
    data class ProductId(val value: Long)

    // Create Db2Type from bigint
    val productIdType: Db2Type<ProductId> = Db2Types.bigint.transform(::ProductId, ProductId::value)
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/db2/DomainType.scala
package dev.typr.foundationssc.docs.db2
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

@SuppressWarnings(Array("unused"))
object DomainType:
  // start
  // Wrapper type
  case class ProductId(value: Long)

  // Create Db2Type from bigint
  val productIdType: Db2Type[ProductId] = Db2Types.bigint.transform(ProductId.apply, _.value)
  // stop
